src/classes/groundplaces.ts

```diff
@@ -25,42 +34,54 @@ export class GroundPlaces {
   public autocomplete(query: string, filters: AutoCompleteFilters[]): AutoComplete {}
 
   /**
-   * @description Create the stopGroup with the values given.
+   * @description Create a new StopGroup from a SegmentProviderStop.
+   * @param {SegmentProviderStop} segmentProviderStop - The SegmentProviderStop on which is based the StopGroup.
+   * @param {StopGroupInfos} stopGroupInfos - StopGroup informations.
    * @param {StopClusterGpuid} stopClusterGpuid - Ground place unique identifier of the stopCluster parent.
-   * @param {StopGroup} stopGroupInfos - StopGroup informations.
-   * @returns {StopGroup}
+   * @returns {void}
    */
-  // @ts-ignore
-  public createStopGroup(stopClusterGpuid: StopClusterGpuid, stopGroupInfos: StopGroup): void {}
+  public createStopGroup(
+    segmentProviderStop: SegmentProviderStop,
+    stopGroupInfos: StopGroupInfos,
+    stopClusterGpuid: StopClusterGpuid,
+  ): void {}
 
   /**
-   * @description Create the stopCluster with the values given.
-   * @param {StopCluster} stopClusterInfos - StopCluster informations.
-   * @returns {StopCluster}
+   * @description Create a new StopCluster from a StopGroup.
+   * @param {StopClusterInfos} stopClusterInfos - StopCluster informations.
+   * @returns {void}
    */
-  // @ts-ignore
-  public createStopCluster(stopClusterInfos: StopCluster): void {}
+  public createStopCluster(stopClusterInfos: StopClusterInfos): void {}
 
   /**
    * @description Update the stopGroup with the new values given.
    * @param {StopGroupGpuid} stopGroupGpuid - Ground place unique identifier of a StopGroup.
    * @param {StopGroupProperties} propertiesToUpdate - Properties that need to be update.
-   * @returns {StopGroup}
+   * @returns {void}
    */
-  // @ts-ignore
-  public updateStopGroup(stopGroupGpuid: StopGroupGpuid, propertiesToUpdate: StopGroupProperties): StopGroup {}
+  public updateStopGroup(stopGroupGpuid: StopGroupGpuid, propertiesToUpdate: StopGroupProperties): void {
+    // Before make call to storageService
+    // Check if the rules are respected with the new properties to update
+    this.checkValidity();
```

Comment:
Is this call will check ALL rules ? 

Isn't it more modulable to create a single and specific check by actions ? (and an other to check all the file)

What do you think @Theo-Le-Donne @Simobain ?

Reply:
I set this up for the moment so that I can check the rules related to this case beforehand. Nothing is implemented here just to get an idea of the structure of the function.

I was thinking about a function where you pass a rule name and the values that are used for that rule.

And the function in question will look at the values in relation to that rule.

Reply:
Two schools there : 
* full check every move 
* specific check every move

Preference for specific check (opti) but could be a first version with a full check.

Reply:
Ok for now I'll go with only one method that check rules and after split it into small pieces!